addons/wowl/static/src/core/download.ts

```diff
@@ -0,0 +1,140 @@
+import download from "../libs/download";
+import { _lt, Stringifiable } from "./localization";
+import parse from "../libs/content-disposition";
+import { Odoo } from "../types";
+
+interface DownloadFileOptionsFromForm {
+  type: "form";
+  form: HTMLFormElement;
+}
+
+interface DownloadFileOptionsFromParams {
+  type: "params";
+  url: string;
+  data: Map<string, string | Blob>;
+}
+
+interface DownloadFileOptionsCallbacks {
+  success?: () => void;
+  error?: (error: {
+    message: Stringifiable | String;
+    data: {
+      name?: Stringifiable | String;
+      title: Stringifiable | String;
+    };
+  }) => void;
+  complete?: () => void;
+}
```

Comment:
I'm not sure about this

Reply:
I also think that error type should be defined more globally, so it mightn't belong here.